CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift

```diff
@@ -136,9 +121,22 @@ public class AnalyticsManager {
     private static func processEventsQueue() {
         while !eventsQueue.isEmpty {
             let queuedEvent = eventsQueue.removeFirst()
-            track(event: queuedEvent.event,
-                  screenNameString: queuedEvent.screenNameString,
-                  properties: queuedEvent.properties)
+            var eventProperties: [String: String] = [:]
+
+            if let screenName = queuedEvent.screenNameString {
+                eventProperties[AnalyticsPropertyKey.screenName.rawValue] = screenName
+            }
+
+            for property in queuedEvent.properties {
+                let propertyValue = property.value.analyticsPropertyValue()
+                eventProperties[property.key.rawValue] = convertPropertyValueToString(propertyValue)
+            }
+
+            // Track event in Mixpanel
+            mixpanelInstance?.track(event: queuedEvent.event.rawValue, properties: eventProperties)
+
+            // Track event in Amplitude
+            ampltitudeTrackEvent(event: queuedEvent.event, eventProperties: eventProperties)
```

Comment:
[suggestion] Maybe we can put this portion of code into a separate method. What do you think?

Reply:
@ValentinaIancu-Gini 
I think we can put this code in method track(event: QueuedAnalyticsEvent) and call it from `processEventsQueue`. In that way we will keep consistent structure of our methods
Works good?